arches/app/models/migrations/10999_add_principaluserid_to_resources.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 2.2.6 on 2019-12-13 11:56
+
+from django.conf import settings
+from django.db import migrations, models, transaction
+from django.db.models import deletion
+
+from arches.app.models.resource import Resource
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("models", "10798_jsonld_importer"),
+    ]
+
+    @transaction.atomic
+    def forward_func(state, editor):
+        for resource in Resource.objects.all():
+            resource.principaluser_id = resource.get_instance_creator()
+            resource.save()
```

Comment:
This is a one-time only thing, so it's probably okay to have an N+1 queries problem here with one editlog query per resource. But we could look into a solution for it, especially if that's a use case we anticipate having in the application: are there cases where we iterate a bunch of resources and should be "joining" the EditLog table instead of querying for every row?

A couple other questions:
- do we need iterator() here in case there are deployments with a very high count of resources?
- should this be atomic? Django [cautions](https://docs.djangoproject.com/en/5.0/howto/writing-migrations/#non-atomic-migrations) that large data migrations can be slow when performed inside transactions

Reply:
Yeah, this is not great.  A ton of resources would probably end up making for a long upgrade time.